src/k2/tso/client/tso_clientlib.cpp

```diff
@@ -87,42 +87,37 @@ seastar::future<> TSO_ClientLib::DiscoverServiceNodes(const k2::String& serverUR
             return seastar::make_exception_future<>(TSOClientLibShutdownException());
         }
 
-        return k2::RPC().sendRequest(dto::Verbs::GET_TSO_SERVICE_NODE_URLS, myRemote->newPayload(), *myRemote, timeout)
-        .then([this](std::unique_ptr<k2::Payload> payload) {
+        GetTSOServiceNodeURLsRequest request;  // empty request param
+
+        return k2::RPC().callRPC<dto::GetTSOServiceNodeURLsRequest, dto::GetTSOServiceNodeURLsResponse>(dto::Verbs::GET_TSO_SERVICE_NODE_URLS, request, *myRemote, timeout)
+        .then([this](auto&& response) {
             if (_stopped) return seastar::make_ready_future<>();
 
-            if (!payload || payload->getSize() == 0)
+            auto& [status, r] = response;
+            if (!status.is2xxOK())
             {
-                K2LOG_E(log::tsoclient, "Remote end did not provide a data endpoint. Giving up");
-                return seastar::make_exception_future<>(std::runtime_error("no remote endpoint"));
+                K2LOG_E(log::tsoclient, "Error during get TSO node URLs, code:{}, error message:{}", status.code, status.message);

```

Comment:
you can just format status here instead of each part:  "nodeURLs, status={}", status)

Reply:
Good to know. Thx!